server/game/cards/13.1-AtG/GateOfTheGods.js

```diff
@@ -0,0 +1,43 @@
+const DrawCard = require('../../drawcard.js');
+const TextHelper = require('../../TextHelper');
+
+class GateOfTheGods extends DrawCard {
+    setupCardAbilities(ability) {
+        this.plotModifiers({
+            gold: 1
+        });
+        this.action({
+            title: 'Sacrifice to draw 2 cards',
+            condition: () => this.characterWithHighestStrength(),
+            phase: 'challenge',
+            cost: ability.costs.sacrificeSelf(),
+            handler: () => {
+                let cards = this.controller.drawCardsToHand(2).length;
+                this.game.addMessage('{0} sacrifices {1} to draw {2}', 
+                    this.controller, this, TextHelper.count(cards, 'card'));
+            }
+        });
+    }
+
+    characterWithHighestStrength() {
+        let charactersInPlay = this.game.filterCardsInPlay(card => card.getType() === 'character');
+        if(charactersInPlay.length === 0) {
+            return false;
+        }
+        let cardWithHighestStrength;
+        let highestStrength = 0;
+        charactersInPlay.forEach(function(card) {
```

Comment:
This logic fails if multiple players tie for the highest STR character. e.g. one player controls Core Randyl Tarly (STR 5), another controls Core Balon (STR 5). The player with Gate of the Gods should be able to trigger it even though there's a tie, and the current code will sometimes allow them to but sometimes not, depending on which player created the game.

The logic should be more along the lines of:
```
get the highest STR in play
get a list of characters that have that STR
return true if the current player controls one of those characters
```

As a code style note, we generally prefer using `for` loops instead of the `forEach` method, unless there's a good reason to do so.